### ISSUE TITLE
no carrot breaks on unpublished modules

### ISSUE THREAD

**Issue by kolodny** [kittens/lerna#93]
For example in `/packages/a/package.json`, if I have `"dependencies": { "b": "1.0.0" }` if `b` is not published it will throw an error:

```
npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/b
```

OTOH if I use `"dependencies": { "b": "^1.0.0" }` then it will work

This is all assuming that `b` is in the packages directory and has a version of `1.0.0`

Something else I noticed was that after I successfully bootstrap with a carrot, I can then remove the carrot and subsequent calls to bootstrap will work

**Comment by kolodny** [comment]
I started trying to play around with this and it seems that the master branch of lerna has this problem with the carrot as well as without. Here's the script to reproduce, I had lerna with `npm link .` as a sibling of this test directory

```
#!/bin/sh

rm -rf node_modules package.json .git packages
npm init -y > /dev/null
git init
npm link lerna
./node_modules/.bin/lerna init
echo 1.0.0 > VERSION
pkg1=p$RANDOM
pkg2=p$RANDOM
mkdir packages/$pkg1
cd packages/$pkg1
npm init -y > /dev/null
cd ../..
mkdir packages/$pkg2
cd packages/$pkg2
npm init -y > /dev/null

cat <<EOF | node
var pkg = require('./package.json');
pkg.dependencies = {$pkg1: '^1.0.0'}
require('fs').writeFileSync('./package.json', JSON.stringify(pkg, null, 2));
EOF

cd ../..

./node_modules/.bin/lerna bootstrap
```

**Comment by thejameskyle** [comment]
Fixed in 2.0.0-beta.4
